cdap-proto/src/main/java/co/cask/cdap/proto/Id.java

```diff
@@ -200,4 +209,144 @@ public String toString() {
       return sb.toString();
     }
   }
+
+  /**
+   * Represents ID of a Notification feed.
+   */
+  public static class NotificationFeed {
+
+    private final Namespace namespace;
+    private final String category;
+    private final String name;
+
+    private final String description;
+
+    /**
+     * {@link NotificationFeed} object from an id in the form of "namespace.category.name".
+     *
+     * @param id id of the notification feed to build
+     * @return a {@link NotificationFeed} object which id is the same as {@code id}
+     * @throws IllegalArgumentException when the id doesn't match a valid feed id
+     */
+    public static NotificationFeed fromId(String id) {
+      String[] idParts = id.split("\\.");
+      if (idParts.length != 3) {
+        throw new IllegalArgumentException(String.format("Id %s is not a valid feed id.", id));
+      }
+      return new NotificationFeed(idParts[0], idParts[1], idParts[2], "");
+    }
+
+    private NotificationFeed(String namespace, String category, String name, String description) {
+      Preconditions.checkArgument(namespace != null && !namespace.isEmpty(),
+                                  "Namespace value cannot be null or empty.");
+      Preconditions.checkArgument(category != null && !category.isEmpty(), "Category value cannot be null or empty.");
+      Preconditions.checkArgument(name != null && !name.isEmpty(), "Name value cannot be null or empty.");
+      Preconditions.checkArgument(isId(namespace) && isId(category) && isId(name),
```

Comment:
Yeah, the concern I have is, since we are exposing the ``isId`` as sharable between all ``Id``s, it would be good if it throws an ``InvalidIdException`` or something similar, which all constructors in the ``Id`` subclasses would propagate, so anytime you create an ``Id.Application`` or ``Id.Program`` or anything else,  it clearly indicates to clients that they need to handle that case.